lib/ansible/modules/cloud/amazon/GUIDELINES.md

```diff
@@ -350,6 +374,32 @@ def describe_some_resource(client, module):
         module.fail_json_aws(e, msg="Could not describe resource %s" % name)
 ```
 
+To make use of AWSRetry easier, it can now be wrapped around a client returned
+by `AnsibleAWSModule`. any call from a client. To add retries to a client,
+create a client:
+
+```
+module.client('ec2', retry_decorator=AWSRetry.jittered_backoff(retries=10))
+```
+
+Any calls from that client can be made to use the decorator passed at call-time
+using the `aws_retry` argument. By default, no retries are used.
+
+```
+ec2 = module.client('ec2', retry_decorator=AWSRetry.jittered_backoff(retries=10))
+ec2.describe_instances(InstanceIds=['i-123456789'], aws_retry=True)
+
+# equivalent with normal AWSRetry
+@AWSRetry.jittered_backoff(retries=10)
+def describe_instances(client, **kwargs):
+    return ec2.describe_instances(**kwargs)
+
+describe_instances(module.client('ec2'), InstanceIds=['i-123456789'])
+```
+
+The call will be retried the specified number of times, so the calling functions
+don't need to be wrapped in the backoff decorator.
```

Comment:
Does this work with pagination too? Presumably that might be

```
paginator = ec2.get_paginator('some_service', aws_retry=True) 
result = paginator.paginate(**params).build_full_result()
```

Reply:
Ah, that's a great point. It doesn't work with pagination (unfortunately) because the path boto3 goes to paginate resources doesn't give us the ability to wrap those calls in the same way we can wrap client calls.